lib/internal/Magento/Framework/Image/Adapter/Gd2.php

```diff
@@ -71,10 +71,12 @@ public function open($filename)
             throw new \OverflowException('Memory limit has been reached.');
         }
         $this->imageDestroy();
-        $this->_imageHandler = call_user_func(
-            $this->_getCallback('create', null, sprintf('Unsupported image format. File: %s', $this->_fileName)),
-            $this->_fileName
-        );
+        if (strpos($this->_fileMimeType, 'ico') === false) {
```

Comment:
Why special `if` for `ico` only?
https://github.com/magento/magento2/pull/25007/files#diff-eb0ebedd963043bcbd2fbce344602c6aR81 has the same purpose.
Same question to other similar `if`s in this file. Can you combine those with logic for other types, so it's all in one place?